server/client.go

```diff
@@ -943,7 +965,30 @@ func (c *client) typeString() string {
 	return "Unknown Type"
 }
 
+func (c *client) closeConnectionWithEvent(reason string) {
+
+	c.mu.Lock()
+
+	if c.nc == nil {
```

Comment:
I agree, and good point.  However, to avoid overhead of a closed flag, perhaps checking the server's client map before publishing would work...  There would still be a small window for duplicate disconnect messages though.

e.g.
```
_, exists := srv.clients[cid]
     if !exists {
        return
    }
```